examples/notebooks/beam-ml/run_inference_windowing.ipynb

```diff
@@ -0,0 +1,477 @@
+        "id": "KNRkuUQ9aA62"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def preprocess_data(\n",
+        "    dataset_path: str,\n",
+        "    training_set_path: str,\n",
+        "    labels_path: str,\n",
+        "    test_set_path: str):\n",
+        "  \"\"\"\n",
+        "    Helper function to split the dataset into a training set\n",
+        "    and its labels and a test set. The training set and\n",
+        "    its labels are used to train a lightweight model.\n",
+        "    The test set is used to create a test streaming pipeline.\n",
+        "    Args:\n",
+        "        dataset_path: path to csv file containing the Kaggle\n",
+        "         milk quality dataset\n",
+        "        training_set_path: path to output the training samples\n",
+        "        labels_path:  path to output the labels for the training set\n",
+        "        test_set_path: path to output the test samples\n",
+        "    \"\"\"\n",
+        "  df = pandas.read_csv(dataset_path)\n",
+        "  df['Grade'].replace(['low', 'medium', 'high'], [0, 1, 2], inplace=True)\n",
+        "  x = df.drop(columns=['Grade'])\n",
+        "  y = df['Grade']\n",
+        "  x_train, x_test, y_train, _ = \\\n",
+        "      train_test_split(x, y, test_size=0.60, random_state=99)\n",
+        "  x_train.to_csv(training_set_path, index=False)\n",
+        "  y_train.to_csv(labels_path, index=False)\n",
+        "  x_test.to_csv(test_set_path, index=False)\n",
+        "\n",
+        "\n",
+        "def train_model(\n",
+        "    samples_path: str, labels_path: str, model_state_output_path: str):\n",
+        "  \"\"\"Function to train the XGBoost model.\n",
+        "    Args:\n",
+        "      samples_path: path to csv file containing the training data\n",
+        "      labels_path: path to csv file containing the labels for the training data\n",
+        "      model_state_output_path: Path to store the trained model\n",
+        "  \"\"\"\n",
+        "  samples = pandas.read_csv(samples_path)\n",
+        "  labels = pandas.read_csv(labels_path)\n",
+        "  xgb = xgboost.XGBClassifier(max_depth=3)\n",
+        "  xgb.fit(samples, labels)\n",
+        "  xgb.save_model(model_state_output_path)\n",
+        "  return xgb"
+      ],
+      "metadata": {
+        "id": "MUUq_j6NXu41"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Preprocess the data and train the model\n",
+        "\n",
+        "We split the dataset in a training set, test set and labels for training set. We use the test set as input data for our test stream, you can use the test set to validate the trained model's performance. After preprocessing, we have 3 different files containing data. Once we have our training set, we can also train the XGBoost model and store it in a json file, such that it can be loaded by the ModelHandler."
```

Comment:
```suggestion
        "Split the dataset into three files with data: a training dataset, a test dataset, and labels for the training dataset. Use the test set as input data for the test stream and to validate the trained model's performance. Use the training set to train the XGBoost model. Store the model in a JSON file so that the model handler can load the model."
```